### ISSUE TITLE
pkg.installed not failing even though apt was reporting missing dependencies

### ISSUE THREAD

**Issue by midihenry** [saltstack/salt#27619]
I had this problem when trying to run this state file:
```salt
postgresql_install:
  pkg.installed:
    - name: postgresql-9.3
    - refresh: True

add_user:
  postgres_user.present:
    - name: user
    - password: user
    - user: postgres
    - require:
      - pkg: postgresql_install
```
Postrgresql fails to install because of missing dependencies found on the minion log:
![minion_log](https://cloud.githubusercontent.com/assets/7854083/10243262/57484a04-68ef-11e5-8683-40e3d236fbad.png)

 But instead of the `postgresql_install` state failing, the `add_user` state is executed and I get a KeyError which is caused by the absence of the psql command (postgresql failed to install in the first place):
```python-traceback
The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1161, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 523, in highstate
        whitelist=kwargs.get('whitelist')
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2968, in call_highstate
        return self.state.call_high(high)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2067, in call_high
        ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1623, in call_chunks
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1769, in call_chunk
        self._mod_init(low)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 612, in _mod_init
        self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
      File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 90, in __getitem__
        raise KeyError(key)
    KeyError: 'postgres_user.present'
```

It may be a bug with the pkg state module.
I'm using salt-master version **2015.5.5** and **2015.5.3** for the salt-minion

**Comment by jfindlay** [comment]
@midihenry, thanks for the report.

**Comment by rallytime** [comment]
@midihenry This looks like a duplicate scenario of #26882 and #26280 and should be fixed by #27681. The fix will be available in 2015.5.6 and is already fixed in 2015.8.0.

**Comment by rallytime** [comment]
Update: It appears that I mis-remembered when my back-porting PR went into the 2015.5 branch and it missed the 2015.5.6 release tag. However, the initial stacktrace reported by @zer0def about the KeyError should be fixed at the HEAD of the 2015.5 branch. The fix is in 2015.8.0, though, so that statement remains true.
